src/SFA.DAS.EAS.Portal.Client/IPortalClient.cs

```diff
@@ -6,6 +6,7 @@ namespace SFA.DAS.EAS.Portal.Client
 {
     public interface IPortalClient
     {
-        Task<Account> GetAccount(long accountId, CancellationToken cancellationToken = default);
+        Task<Account> GetAccount(string hashedAccountId, AccountState accountState,
```

Comment:
This interface is a query interface.  Basically we are asking for an account filtered by some data.  AccountState with 
 a HasPayeScheme property is a bit meaningless in this instance.  What the client really wants is to get vacancy data included in the resultset.  Think of it as doing a search and limiting or expanding the results of the search.  The service just needs to be told what to return.  A filter with an "IncludeVacancyData = true" property or similar is more intuitive to a client than "HasPayeScheme = true".  In the context of a Get request HasPayeScheme isn't a clear request without knowing about a business rule that may change.  As a client I would expect HasPayeScheme to return the same data whether true or false with the difference being that when true only an account with a PayeScheme would be returned/filtered.

Reply:
I agree with IncludeVacancyData > HasPayeScheme. I think this is a fair point

Reply:
The reason I went with `hasPayeScheme` over e.g. `includeVacancyData`, is that we can potentially reuse `hasPayeScheme` to apply other optimisations, based on whether the account has a paye scheme or not. If we go with instructional flags, (such as `includeVacancyData`), then we could end up sending multiple flags, that are all either set to true or false as a group (effectively rendering all but one of the flags redundant).

It also puts the logic in the client (which I think of as a service for the website), rather than in the website/controller.

The flags enum is named `AccountState` to indicate that that it represents the state of the account. If it was a filter as you expected it to be, I would have called it something like `FilterBy`. I've added XML documentation to the client's public interface, to describe the purpose of the parameters, which intellisense should pop up.

Reply:
I think the point still stands here that the consumer of our client doesn't necessarily know what they are getting or omitting by setting the account state whereas a filter of IncludeVacancyData would be more explicit

Reply:
That's the beauty of encapsulation, the client doesn't need to know the details of why it gets everything it needs, it just does. I can change the flag's name to `IncludeVacancies`, or whatever you wish, if that's your call.

Reply:
I just think that coming at it as a new dev you wouldn't have a clue why you were getting no vacancy data and that you need to know the business rule of "HasPayeScheme must be true in order to get Vacancy Data"

Reply:
As long as the the client passes the correct state of the account, they will get vacancy data (if there is any). The client _does not_ need to know the business rule, if the rule is contained within the client. They _will_ need to know the business rule, if it's contained in the client!

Reply:
Actually, if the client passes the maximum number of vacancies it wants, there's no needs to pass a flag, it can just say it want 0 vacancies.

Reply:
If the business rule changes though then both the caller and the portal client would need to be changed to set/read new setting rather than simply changing where IncludeVacancyData is set on the consumer end?

Reply:
Moving the business logic up the stack, does mean that any change to the logic has the possibility of impacting more of the stack.

However, in this case, I think that any code changes required from the business logic changing, should be able to be contained to the website. If it requires vacancy records, it can say how many it wants, and if it doesn't need any, it can pass 0, with no changes to the portal client required.

Reply:
The interface has now been agreed.